mailpoet/lib/Subscription/Registration.php

```diff
@@ -87,6 +87,7 @@ public function onRegister(
       empty($errors->errors)
       && isset($_POST['mailpoet']['subscribe_on_register'])
       && (bool)$_POST['mailpoet']['subscribe_on_register'] === true
+      && !empty($userEmail)
```

Comment:
I am confused here about the method signature, it says that the email argument is optional. So, is there a valid scenario when the email here can be null?

Also, should we do this check in the previous method `onMultiSiteRegister` as well?

Reply:
> I am confused here about the method signature, it says that the email argument is optional. So, is there a valid scenario when the email here can be null?

@pavel-mailpoet I was not able to find why exactly this parameter can be skipped. It is a very old code. I see that we [register the callback on `registration_erros` filter](https://github.com/mailpoet/mailpoet/blob/59733db6deb51dbf9563af4bacea0f569e8ad474/mailpoet/lib/Config/Hooks.php#L208-L214) so perhaps in some cases in may come null when there are errors. But the Fatal error happens when there are no registration errors but the email was an empty string. That's what I see in the stack trace.

I added the check also to the onMultisiteRegister.